tools/src/main/scala/caliban/tools/ClientWriter.scala

```diff
@@ -232,12 +232,21 @@ object ClientWriter {
             )
           }
         } else {
-          (
-            s"[$typeLetter]",
-            s"(innerSelection: SelectionBuilder[$fieldType, $typeLetter])",
-            writeType(field.ofType).replace(fieldType, typeLetter),
-            writeTypeBuilder(field.ofType, "Obj(innerSelection)")
-          )
+          if (fieldType == "Option") {
+            (
+              s"[$typeLetter]",
+              s"(innerSelection: SelectionBuilder[$fieldType, $typeLetter])",
+              writeType(field.ofType).reverse.replaceFirst(fieldType.reverse, typeLetter).reverse,
```

Comment:
How about moving this line to a separation function, with the `if (fieldType == "Option")` in that function, so that the 3 other lines are not duplicated?

Reply:
:100: I added a `safeFieldTypeReplace` method to handle this